nemo/collections/tts/waveglow_modules.py

```diff
@@ -207,16 +207,15 @@ def denoise(self, audio, strength=0.1):
         audio_denoised = librosa.core.istft(audio_spec_denoised * audio_angles)
         return audio_denoised, audio_spec_denoised
 
-    def forward(self, mel_spectrogram):
+    def forward(self, mel_spectrogram, z):
         if not self._removed_weight_norm:
             logging.info("remove WN")
-            self.waveglow = self.waveglow.remove_weightnorm(self.waveglow)
+            self.waveglow = remove_weightnorm(self.waveglow)
             self._removed_weight_norm = True
         if self.training:
             raise ValueError("You are using the WaveGlow Infer Neural Module in training mode.")
         with torch.no_grad():
-            audio = self.waveglow.infer(mel_spectrogram, sigma=self._sigma)
-        return audio
+            return self.waveglow.forward((mel_spectrogram, z,))[0]
```

Comment:
@drnikolaev, why did you add z to the forward method of `WaveglowInferNM` and change it from waveglow.infer to waveglow.forward?

Reply:
@titu1994

Reply:
@drnikolaev Is this required for ONNX or TRT? If its for conversion compat, can I add a flag that uses infer by default, but switches to this forward? Also, the input_ports needs to be modified to support the `z` variable in such a case.

Reply:
It was made to avoid tracer error which was unrecoverable. You can try to undo it and see yourself.

Reply:
So will the flag technique support both use cases ? Or will tracer (is this the JIT trace of pytorch?) complain that execution paths have different inputs and outputs ? Also, how do we supply `z` to the neural module as it is currently ?

Reply:
I'll need some time to reproduce the error, please hold on